plugins/plugin-chart-echarts/src/Graph/transformProps.ts

```diff
@@ -0,0 +1,178 @@
+    const symbolSize = node.symbolSize;
+    if (symbolSize > max) {
+      max = symbolSize;
+    }
+    if (symbolSize < min) {
+      min = symbolSize;
+    }
+  });
+
+  nodes.forEach((node: { symbolSize: number }) => {
+    node.symbolSize = ((node.symbolSize - min) / (max - min)) * 60 + 10;
+  });
+}
+
+export default function transformProps(chartProps: ChartProps): EchartsProps {
+  const { width, height, formData, queriesData } = chartProps;
+  const data: DataRecord[] = queriesData[0].data || [];
+
+  const {
+    name,
+    source,
+    target,
+    category,
+    colorScheme,
+    metric = '',
+    layout,
+    roam,
+    draggable,
+    selectedMode,
+    showSymbolThreshold,
+  }: EchartsGraphFormData = { ...DEFAULT_GRAPH_FORM_DATA, ...formData };
+
+  const metricLabel = getMetricLabel(metric);
+  const colorFn = CategoricalColorNamespace.getScale(colorScheme as string);
+  let nodes: { [name: string]: number } = {};
+  let echart_nodes: {
+    id: number;
+    name: DataRecordValue;
+    symbolSize: any;
+    value: any;
+    label?: { [name: string]: boolean };
+    category: string | null;
+  }[] = [];
+  let echart_links: object[] = [];
+  let echart_categories: string[] = [];
+  let index = 0;
+  let source_index = 0;
+  let target_index = 0;
+
+  data.forEach(link => {
+    const node_source: any = link[source];
+    const node_target: any = link[target];
+    const node_category: string = category ? link[category]!.toString() : 'default';
+
+    if (!(node_source in nodes)) {
+      echart_nodes.push({
+        id: index,
+        name: node_source,
+        value: link[metricLabel],
+        symbolSize: link[metricLabel],
+        category: node_category,
+      });
+      source_index = index;
+      nodes[node_source] = index;
+      index += 1;
+    } else {
+      source_index = nodes[node_source];
+      echart_nodes[source_index].value += link[metricLabel];
+      echart_nodes[source_index].symbolSize += link[metricLabel];
+    }
+
+    if (!(node_target in nodes)) {
+      echart_nodes.push({
+        id: index,
+        name: node_target,
+        value: link[metricLabel],
+        symbolSize: link[metricLabel],
+        category: node_category,
+      });
+      target_index = index;
+      nodes[node_target] = index;
+      index += 1;
+    } else {
+      target_index = nodes[node_target];
+      echart_nodes[target_index].value += link[metricLabel];
+      echart_nodes[target_index].symbolSize += link[metricLabel];
+    }
+    echart_links.push({ source: source_index.toString(), target: target_index.toString() });
+
+    if (!echart_categories.includes(node_category)) {
+      echart_categories.push(node_category);
+    }
+  });
+  if (showSymbolThreshold > 0) {
+    echart_nodes.forEach(function (node) {
+      node.label = {
+        show: node.value > showSymbolThreshold,
+      };
+    });
+  }
+
+  normalizeSymbolSize(echart_nodes);
+
+  const echartOptions: echarts.EChartOption = {
```

Comment:
With #928 we can now use types provided by the ECharts package. You check how these have been implemented in the other ECharts based viz plugins.